src/main/webapp/scripts/user-script.js

```diff
@@ -118,14 +119,53 @@ function getAllUsers(){
         }
         }).
         then(response => response.json())
-        .then((ous) => {
-            console.log(ous);
+        .then((userJSON) => {
+            let users = userJSON['users'];
+            for(var i = 0; i < users.length; i++){
+                let user = users[i];
+                var fullname = user['name']['fullName'];
+                var id = user['id'];
+                var orgUnitPath = user['orgUnitPath'];
+                var userJSON = {"name": fullname, "id": id, "orgUnitPath": orgUnitPath};
+                DFSTree(data, orgUnitPath, userJSON);
+            }
+            addValue(data);
+            visualize();
         })
         .catch((error) => {
             console.error('Error:', error);
         });
 }
 
+// add user to the OU it's in by DFS 
+function DFSTree(node, path, user){
+    if(node['path'] === path){
+        node['users'].push(user);
+        return;
+    }
+    else{
+        var children = node['children'];
+        for(var i = 0; i < children.length; i++){
+            ou = children[i];
+            if(path.includes(ou['path'])){
+                DFSTree(ou, path, user);
+            }
+        }
+    }
+}
+
+// change value of each OU to the number of users in the OU
+function addValue(node){
```

Comment:
Maybe rename the function to incrementUserCount ?

Reply:
Sure! Changed in new commit